scripts/git-branch.ps1

```diff
@@ -0,0 +1,135 @@
+
+# create a new release branch with annoated tag and push to git origin
+
+[CmdletBinding()]
+param
+(
+    [Parameter(Mandatory = $false)]
+    [string]$Name,
+    
+    [Parameter(Mandatory = $false)]
+    [string]$Email
+)
+
+function TagExists($TagName, [switch]$Remote) {
+    
+    # look for the name in local refs/tags
+
+    if ($Remote -eq $true) {
+
+        foreach ($tag in $(git ls-remote --tags)) {
+
+            $name = Split-Path $tag -Leaf
+    
+            if ($name -eq $TagName) {
+                return $true
+            }
+        }
+    }
+    else {
+
+        foreach ($tag in $(git tag)) {
+
+            if ($tag -eq $TagName) {
+                return $true
+            }
+        }
+    }
+
+    return $false
+}
+
+function ExitOrContinue {
+
+    # stop only if prior git command failed
+
+    Write-Host "Last exit code is ($LASTEXITCODE)"
+
+    if ($LASTEXITCODE -ne 0) {
+        exit $LASTEXITCODE  # terminate
+    }
+}
+
+function CurrentBranch() {
+
+    # find current branch name from git output
+
+    foreach ($line in $(git branch)) {
+
+        # git branch output marks the current branch with a leading astrisk
+
+        if ($line.StartsWith('*')) {
+            return $line
+        }
```

Comment:
nit: I kinda liked having the trimming of the spaces and * here, makes the code below cleaner. no big deal but if it's easy to bring back... i think it's worth it.